cpp/src/reductions/scan/rank_scan.cu

```diff
@@ -51,20 +51,34 @@ std::unique_ptr<column> rank_generator(column_view const& order_by,
                                        rmm::cuda_stream_view stream,
                                        rmm::mr::device_memory_resource* mr)
 {
-  auto comp = cudf::experimental::row::equality::self_comparator(table_view{{order_by}}, stream);
-  auto const device_comparator =
-    comp.equal_to(nullate::DYNAMIC{has_nested_nulls(table_view({order_by}))});
+  auto const order_by_view = table_view{{order_by}};
+  auto comp = cudf::experimental::row::equality::self_comparator(order_by_view, stream);
+
   auto ranks = make_fixed_width_column(
     data_type{type_to_id<size_type>()}, order_by.size(), mask_state::UNALLOCATED, stream, mr);
   auto mutable_ranks = ranks->mutable_view();
 
-  thrust::tabulate(rmm::exec_policy(stream),
-                   mutable_ranks.begin<size_type>(),
-                   mutable_ranks.end<size_type>(),
-                   [comparator = device_comparator, resolver] __device__(size_type row_index) {
-                     return resolver(row_index == 0 || !comparator(row_index, row_index - 1),
-                                     row_index);
-                   });
+  if (cudf::detail::has_nested_columns(order_by_view)) {
+    auto const device_comparator =
+      comp.equal_to<true>(nullate::DYNAMIC{has_nested_nulls(table_view({order_by}))});
+    thrust::tabulate(rmm::exec_policy(stream),
+                     mutable_ranks.begin<size_type>(),
+                     mutable_ranks.end<size_type>(),
+                     [comparator = device_comparator, resolver] __device__(size_type row_index) {
+                       return resolver(row_index == 0 || !comparator(row_index, row_index - 1),
+                                       row_index);
+                     });
+  } else {
+    auto const device_comparator =
+      comp.equal_to<false>(nullate::DYNAMIC{has_nested_nulls(table_view({order_by}))});
+    thrust::tabulate(rmm::exec_policy(stream),
+                     mutable_ranks.begin<size_type>(),
+                     mutable_ranks.end<size_type>(),
+                     [comparator = device_comparator, resolver] __device__(size_type row_index) {
+                       return resolver(row_index == 0 || !comparator(row_index, row_index - 1),
+                                       row_index);
```

Comment:
Refactor as functor. Perhaps add documentation (to the functor) on the logic here.

Reply:
If you like, I can file an issue for the same. I haven't looked closely at the logic itself, and only done restructuring where needed

Reply:
File an issue for what exactly? I think refactoring this to a functor falls under this PR. 
Maybe you are referring to adding documentation?

Reply:
Yes, I'm talking about the documentation indeed.

Reply:
The documentation is not necessary. The refactoring looks good to me.